gen/__init__.py

```diff
@@ -306,20 +305,82 @@ def do_gen_package(config, package_filename):
     log.info("Package filename: %s", package_filename)
 
 
-def extract_files_with_path(start_files, paths):
+def render_late_content(content, late_values):
+    # TODO(branden): Unit tests
+
+    def _next_substring(string, substring, start=0):
```

Comment:
In this module, consider replacing the variable name string with something else, even _string alright.
There is a module that string that this variable can shadow. If we ever happen to use it, better to use the module as string than import it as _string.

Changing this variable name could be better.